healthchain/use_cases/cds.py

```diff
@@ -150,31 +155,34 @@ def cds_service(self, id: str, request: CDSRequest) -> CDSResponse:
         """
         CDS service endpoint for FastAPI app, should be mounted to /cds-services/{id}
         """
+        # TODO: can register multiple services and fetch with id
         if self.service_api is None:
             log.warning("CDS 'service_api' not configured, check class init.")
             return CDSResponse(cards=[])
 
-        # TODO: can register multiple services and fetch with id
-        request_json = request.model_dump_json(
-            exclude_none=True, exclude_unset=True, exclude_defaults=True, by_alias=True
-        )
         signature = inspect.signature(self.service_api.func)
         assert (
             len(signature.parameters) == 2
         ), f"Incorrect number of arguments: {len(signature.parameters)} {signature}; CDS Service functions currently only accept 'self' and a single input argument."
 
-        # TODO: better handling of args/kwargs io
-        # params = iter(inspect.signature(self._service_api.func).parameters.items())
-        # for name, param in params:
-        #     print(name, param, param.annotation)
+        service_input = request
+        params = iter(inspect.signature(self.service_api.func).parameters.items())
+        for name, param in params:
+            if name != "self":
+                if param.annotation == str:
+                    service_input = request.model_dump_json(exclude_none=True)
+                elif param.annotation == Dict:
+                    service_input = request.model_dump(exclude_none=True)
 
-        result = self.service_api.func(self, request_json)
+        result = self.service_api.func(self, service_input)
 
         # TODO: could use llm to check and fix results here?
         if result is None:
             log.warning(
                 "CDS 'service_api' returned None, please check function definition."
             )
             return CDSResponse(cards=[])
+        elif isinstance(result, Card):
+            result = [result]
```

Comment:
You probably want to add an extra statement to catch the case where result is neither None nor a Card.